plugins/inputs/oomkiller/oomkiller.go

```diff
@@ -0,0 +1,75 @@
+package oomkiller
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/influxdata/tail"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Oomkiller struct {
+	Logfile string `toml:"logfile"`
+}
+
+var sampleConfig = `
+  ## Logfile where oom killer is reflected
+  logfile = "/var/log/messages"
+`
+
+// SampleConfig returns a sample configuration block
+func (a *Oomkiller) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description just returns a short description of the Mesos plugin
+func (a *Oomkiller) Description() string {
+	return "Telegraf plugin for gathering metrics from kernel Oomkiller"
+}
+
+func (a *Oomkiller) SetDefaults() {
+	if a.Logfile == "" {
+		log.Println("I! [oomkiller] Missing logfile value, setting default value (/var/log/messages)")
+		a.Logfile = "/var/log/messages"
+	}
+}
+
+func IsOomkillerEvent(line string) bool {
+	matched, _ := regexp.MatchString("invoked oom-killer", line)
+	return matched
+}
+
+// Gather() metrics from Oomkiller
+func (a *Oomkiller) Gather(acc telegraf.Accumulator) error {
+	a.SetDefaults()
+	//This is to NOT start tailing log from the beginning
+	var seek *tail.SeekInfo
+	seek = &tail.SeekInfo{Whence: 2, Offset: 0}
+	t, err := tail.TailFile(a.Logfile, tail.Config{Follow: true, Location: seek, ReOpen: true})
+	if err != nil {
+		return fmt.Errorf("Error: %v", err)
```

Comment:
Ig you want, print a message, but instead of returning a `string` return an `error` type.

Reply:
fixed on https://github.com/medallia/telegraf/pull/16/commits/c8c717880418399d7ebd008d4340173b05c39837